source/credentials_provider_ecs.c

```diff
@@ -629,3 +628,93 @@ struct aws_credentials_provider *aws_credentials_provider_new_ecs(
 
     return NULL;
 }
+
+struct aws_credentials_provider *aws_credentials_provider_new_ecs_from_environment(
+    struct aws_allocator *allocator,
+    const struct aws_credentials_provider_ecs_environment_options *options) {
+
+    struct aws_credentials_provider_ecs_options explicit_options = {
+        .shutdown_options = options->shutdown_options,
+        .bootstrap = options->bootstrap,
+        .function_table = options->function_table,
+    };
+
+    struct aws_string *relative_uri_str = NULL;
+    struct aws_string *full_uri_str = NULL;
+    struct aws_uri full_uri;
+    AWS_ZERO_STRUCT(full_uri);
+    struct aws_credentials_provider *provider = NULL;
+
+    /* Read the auth token variables */
+    struct aws_string *ecs_env_token_file_path = NULL;
+    struct aws_string *ecs_env_token = NULL;
+    if (aws_get_environment_value(allocator, s_ecs_creds_env_token_file, &ecs_env_token_file_path) != AWS_OP_SUCCESS ||
+        aws_get_environment_value(allocator, s_ecs_creds_env_token, &ecs_env_token) != AWS_OP_SUCCESS) {
```

Comment:
trivial/annoying: `ecs_env_token_file_path` would leak if the 1st call was AWS_OP_SUCCESS, but the 2nd call was AWS_OP_ERR

But the funny thing is `aws_get_environment_value()` can't possibly fail (now that allocations can't fail)

Either properly handle each individual error, or just call aws_get_environment_value() as if it didn't return an error code

Reply:
Updated the code ignore the return value from aws_get_environment_value as it makes the code simpler.